### DISCUSSION TITLE
want to parse versiondata from contentversion to create records

### DISCUSSION THREAD

**Discussion by psw-syboo** [jsforce/jsforce#1472] [Q&A]
i've check the following posts but they are all trying to download a file
https://github.com/jsforce/jsforce/issues/19
https://github.com/jsforce/jsforce/issues/471
https://github.com/jsforce/jsforce/issues/553

but i am trying to get the versiondata from the contentversion back to original csv form so i can parse it, not save it to a file since i am running from a browser as a client.

here's my code
    var contentStream = sfConnection.sobject('ContentVersion').record('XXXXXXX').blob('VersionData');
    var buf = [];
    contentStream.on('data', function(a,b,c,d) {
      console.log('ondata',a,b,c,d);
      buf.push(data.toString(enc));
    });
    contentStream.on('end', function() {
      var contentStr = buf.join('');
      console.log('onend',buf,contentStr,contentStream)
    });

when i run this i can see in network that i am downloading the file but i am never getting the 'data' or 'end' events to fire.  i noticed there is also a finished event which does seem to trigger, but since 'data' is not i have no buffered data to be able to decode then parse.  In many posts people say they used connection.request instead of blob but i do not understand how they did it since there was no code example.

general idea, via the web, i want someone to view what files have been uploaded to an account and then choose to parse (extract) them or not.  by doing so they will essentially create X amount of client records linked to the account they uploaded the csv file to (mass import clients via csv by end users).

any help would be appreciated and thanks in advance

**Comment by roshanreacts** [comment]
Same here... Any solutions? It just keeps on writing dead slow... and never comes to end.
